numpy/core/src/multiarray/getset.c

```diff
@@ -931,6 +931,13 @@ array_transpose_get(PyArrayObject *self, void *NPY_UNUSED(ignored))
     return PyArray_Transpose(self, NULL);
 }
 
+static PyObject *
+array_matrix_transpose_get(PyArrayObject *self, void *NPY_UNUSED(ignored))
+{
+    int ndim = PyArray_NDIM(self);
+    return PyArray_SwapAxes(self, ndim - 2, ndim - 1);
```

Comment:
Even if it raises, I expect we should have an explicit error anyway for clarity.

Reply:
I'm not sure if an explicit error is the way to go. Without any changes, `.mT` returns the same as `.T` which is what we want, right? I was thinking along the lines of making it explicit (if `ndim == 1` return `self`).

Reply:
no, it has to error `ndim < 2`, otherwise it gives the wrong idea in formulas.  Maybe adding a dimension might work, but that would require a lot of convincing, so I think it's a no-go right now, to be clear, I am not convinced yet.  (And I don't actually think it is that clear whether someone wants a column or row vector even if we have an implicit decision anyway.)

Reply:
Ok. I'll follow your advice. Would a `ValueError` be a good fit here? I've had a quick look through the available errors and couldn't find anything that fits better.

Reply:
I think that is what we use for shape related errors always, so that sounds good to me.  We were discussing introducing a shape or broadcast error, but that is a different project.